### PR TITLE
CI in buildkite

### PR DESCRIPTION
**Pull request by craigds**
CI implementation using docker containers in buildkite.

Some tweaks were required for the tests because the alpine builds of python don't include the `test` stdlib module. Arguably, we shouldn't have been using it anyway (see the [warning in the docs](https://docs.python.org/3.6/library/test.html))

The CI doesn't do any deployment, just:
* runs tests under python 2.7, 3.5 and 3.6 
* builds the docs, zips the html, and stores the zip as an artifact on buildkite

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+steps:
+  - label: build docs
+    command:
+      - "pip install . && pip install -r requirements-dev.txt && make zipdocs"
```
Comment (rcoup) [inline .buildkite/pipeline.yml]: Shouldn't `pip install .` come _after_ the requirements are installed?

Comment (rcoup) [reply .buildkite/pipeline.yml]: Can split into multiple entries in the `command:` list rather than using `&&`

File: circle.yml [hunk 2]
```diff
@@ -1,28 +0,0 @@
...
-    - cd ${CIRCLE_ARTIFACTS}; zip -r docs.zip docs
-
-test:
-  pre:
-    - mkdir -p ${CIRCLE_TEST_REPORTS}/junit ${CIRCLE_ARTIFACTS}/coverage
-  override:
-    - tox --
-      --verbose
-      -p no:sugar
-      --cov=koordinates
```
Comment (rcoup) [inline circle.yml]: can we keep the coverage report too? Can be [linked from the annotations](https://buildkite.com/docs/agent/v3/cli-annotate#embedding-and-linking-artifacts-in-annotations)

Comment (craigds) [reply circle.yml]: I think that would be only marginally useful. No one will click it, and if you do then you have to click around to compare the percentages between builds.

More useful if there's a big fat change% number somewhere (preferably on github PRs saying "The coverage decreased by 10%"). I couldn't find an easy way to do that from buildkite.

I'm sure I could mess around with writing my own custom annotate, but it's more useful to post to github than to store in CI anyway. 

One option is to sign up to codecov, which is free for OSS and has an upload tool which supports buildkite, though that's not immediately obvious (see https://github.com/codecov/codecov-bash/pull/115). 

It annotates the github PR as a status check: https://docs.codecov.io/docs/commit-status

**Pull request merged**